src/special_arrays/maybe_hot_vector.jl

```diff
@@ -64,7 +64,13 @@ Zygote.@adjoint A::AbstractMatrix * b::MaybeHotVector = (_check_mul(A, b); Zygot
 _mul(A::AbstractMatrix, b::MaybeHotVector{Missing}) = fill(missing, size(A, 1))
 _mul(A::AbstractMatrix, b::MaybeHotVector{<:Integer}) = A[:, b.i]
 
+# this is a bit shady because we're overloading unexported method not intended for public use
+Flux._fast_argmax(x::MaybeHotVector{<:Integer}) = x.i
+# this returns different type
+Flux._fast_argmax(x::MaybeHotVector{<:Maybe{Integer}}) = x.i
+
 Flux.onehot(x::MaybeHotVector{<:Integer}) = Flux.onehot(x.i, 1:x.l)
+maybecold(x::MaybeHotVector{<:Maybe{Integer}}, labels = 1:length(y)) = ismissing(x.i) ? x.i : labels[argmax(x)]
```

Comment:
`labels = 1:length(y)` should probably be `labels = 1:length(x)`.

Also, we do not specialize maybe hot vectors for `argmax`, this will fallback to some generic definition that collects the vector using `getindex` on every element. I would define something like this:

```julia
maybecold(x::MaybeHotVector{Missing}, labels = 1:length(x)) = missing
maybecold(x::MaybeHotVector{<:Integer}, labels = 1:length(x)) = labels[x.i]
```

Reply:
Good catch, thanks.